### PR TITLE
feat: Initial draft for load test scripts

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context

Screwdriver API response needs to be tested for load and gradual ramp up of traffic

## Objective

This PR adds artillery script which help run load against any environment via the pipeline

## Reference
https://github.com/screwdriver-cd/screwdriver/issues/2233

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,29 +1,24 @@
-BSD 3-Clause License
-
-Copyright (c) 2020, Screwdriver
+Copyright Verizon Media
```
Comment (tkyi) [inline LICENSE:1]: should update https://github.com/screwdriver-cd/generator-screwdriver/blob/master/app/templates/LICENSE eventually

File: README.md [hunk 2]
```diff
@@ -1,2 +1,24 @@
 # raptor
 Load test scripts for Screwdriver API
+
+
+# running the load test
+1. Set the env variables $SD_TOKEN, $PIPLINE_ID
```
Comment (tkyi) [inline README.md]:
```suggestion
1. Set the env variables $SD_TOKEN, $PIPELINE_ID
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "raptor",
```
Comment (tkyi) [inline package.json]: might be better to name screwdriver-raptor to fit in naming conventions for our other packages (see https://github.com/screwdriver-cd/models/blob/master/package.json#L2)

```suggestion
  "name": "screwdriver-raptor",
```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/screwdriver-cd/raptor.git"
+  },
+  "author": "Pritam Paul <pritamstyz4ever@gmail.com>",
+  "license": "BSD-3-Clause",
+  "bugs": {
+    "url": "https://github.com/screwdriver-cd/raptor/issues"
```
Comment (tkyi) [inline package.json]:
```suggestion
    "url": "https://github.com/screwdriver-cd/screwdriver/issues"
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,24 @@
 # raptor
 Load test scripts for Screwdriver API
+
+
+# running the load test
+1. Set the env variables $SD_TOKEN, $PIPELINE_ID
```
Comment (jithine) [inline README.md]: SD_TARGET_ENDPOINT missing

File: screwdriver.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+    run-test:
+        requires: [main]
+        steps:
+            - script: |
+                export SD_TARGET_ENDPOINT=$(meta get parameters.SD_TARGET_ENDPOINT.value)
+                export SD_PIPELINE_ID=$(meta get parameters.SD_PIPELINE_ID.value)
+                export SD_API_TOKEN=$(meta get parameters.SD_API_TOKEN.value)
+                RESULT=$(curl $SD_TARGET_ENDPOINT/v4/auth/token?api_token=$SD_API_TOKEN)
+                export SD_TOKEN=$(echo $RESULT | jq '.token')
```
Comment (tk3fftk) [inline screwdriver.yaml]: `node:12` seems not to have `jq`
```
$ docker run -ti --rm node:12 bash -c "jq"
bash: jq: command not found
```

File: screwdriver.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+            screwdriver.cd/ram: HIGH
+            screwdriver.cd/cpu: HIGH
+        requires:
+            - ~pr
+            - ~commit
+        steps:
+            - install: npm install
+
+    run-test:
+        requires: [main]
```
Comment (tk3fftk) [inline screwdriver.yaml:19]: `setup`?

File: apis.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
+## API callsz
```
Comment (tkyi) [inline apis.md]:
```suggestion
## API calls
```

File: report.html [hunk 9]
```diff
@@ -0,0 +1,3702 @@
+{
```
Comment (tkyi) [inline report.html]: are we always planning to check in this file?

Comment (pritamstyz4ever) [reply report.html]: Nope, deleted.

File: screwdriver.yaml (hunk 6 above)
Comment (pritamstyz4ever) [reply screwdriver.yaml]: removed jq!!

File: report.html (hunk 9 above)
Comment (tkyi) [reply report.html]: can you add to the gitignore file

**Review by tkyi — APPROVED** [review]

**Pull request merged**